client/app/src/main/java/com/example/criengine/Activities/RegisterAccount.java

```diff
@@ -100,16 +109,48 @@ protected void onCreate(Bundle savedInstanceState) {
             @Override
             public void onClick(View v) {
+                submitButton.setEnabled(false);
+                username = usernameField.getText().toString();
                 email = emailField.getText().toString();
                 password = passwordField.getText().toString();
                 firstName = firstnameField.getText().toString();
                 lastName = lastnameField.getText().toString();
-
-                // TODO: Create new account in database and log into it.
-
-                // Proceed into the app.
-                Intent intent = new Intent(v.getContext(), RootActivity.class);
-                v.getContext().startActivity(intent);
+                FirebaseFirestore.getInstance().collection("users").whereEqualTo("username",username).get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                        if (task.isSuccessful()) {
+                            QuerySnapshot query = task.getResult();
+                            if (query != null && query.isEmpty()) {
+                                mAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                                    @Override
+                                    public void onComplete(@NonNull Task<AuthResult> task) {
+                                        if (task.isSuccessful() && task.getResult() != null) {
+                                            AuthResult authResult = task.getResult();
+                                            FirebaseUser user = authResult.getUser();
+                                            Profile profile = new Profile(user.getUid(), email, username, firstName, lastName);
+                                            DatabaseWrapper dbw = new DatabaseWrapper(user);
+                                            dbw.addProfile(profile).addOnCompleteListener(new OnCompleteListener<Void>() {
+                                                @Override
+                                                public void onComplete(@NonNull Task<Void> task) {
+                                                    Intent intent = new Intent(v.getContext(), RootActivity.class);
+                                                    v.getContext().startActivity(intent);
+                                                }
+                                            });
+                                        } else {
+                                            Log.d(TAG, "Get Failure: " + task.getException());
+                                            return;
+                                        }
+                                    }
+                                });
+                            } else {
+                                usernameField.setError("Username must be unique");
```

Comment:
Do we need an error tag for the email field too? Or are we allowing multiple people to use the same email?